src/libaktualizr/uptane/tuf.h

```diff
@@ -259,14 +260,43 @@ class Target {
   bool IsOstree() const;
 
   bool operator==(const Target &t2) const {
-    // if (type_ != t2.type_) return false; // Director doesn't include targetFormat
+    // type_ (targetFormat) is only provided by the Images repo.
+    // ecus_ is only provided by the Images repo.
+    // correlation_id_ is only provided by the Director.
+    // uri_ is unchecked because Uptane mentions it should be provided by the
+    // Director, although it can be provided by the Image repository as well.
     if (filename_ != t2.filename_) {
       return false;
     }
     if (length_ != t2.length_) {
       return false;
     }
 
+    // If the HWID vector and ECU->HWID map match, we're good. Otherwise, assume
```

Comment:
It's maybe a naive question, but maybe it's the time where it's too much to have all this logic in an overloaded comparison operator and instead have a dedicated function? The potential for misuse and confusion seems big.

Reply:
Would you prefer to have `operator==` call that function, or to reserve `operator==` only for exact matches and use the new match function for Uptane purposes? I'm hoping the former, because otherwise I'd worry about accidentally doing the wrong thing (using equality instead of the matching function).

Reply:
I imagined that `==` would not call that but then you're probably right. In this case it would be better to delete the comparison operator and only keep the method.
But I'm ready to debate that.

Reply:
Deleting the comparison operator is an interesting idea, but at that point, aren't we back to just having the same thing, just named differently? What's the benefit?

Reply:
As a matter of style it can be surprising to have this complicated business logic inside this operator with a benign name. And the objects are not really "equal", they "match" in some application sense, so I think it's stretching the sense a bit. Also, it makes it hard to grep for usage of this comparison.

It's why I would prefer this to have a scary name but maybe it's not really feasible or we can argue that we contain the usage effectively.

Reply:
I see your points. I think if we delete the == operator then it's probably a reasonable way to go.